packages/browser/src/core/events/interfaces.ts

```diff
@@ -106,13 +116,9 @@ export interface SegmentEvent {
   category?: string
   name?: string
 
-  properties?: object & {
-    [k: string]: JSONValue
-  }
+  properties?: EventProperties
```

Comment:
This might be too strict - we have some users passing in an instance of a class as event properties and TypeScript will complain that the class doesn't have the index signature.

Reply:
I don’t think the typescript types should support people passing “any class”, since it’s not clear how distinguish between a Function and an anonymous class. This is not a breaking change for anyone but potentially typescript users who are doing things I don’t think we should support on the first place.

I think we should write the typescript  API to encourage people to do the right way, i.e do something like {…new MyClass()} (if they must) or have a toJSON() or manually construct an object literal.